python/ray/data/impl/plan.py

```diff
@@ -186,15 +202,16 @@ def _rewrite_read_stage(self) -> Tuple[BlockList, "Stage"]:
         """
         # Generate the "GetReadTasks" stage blocks.
         remote_args = self._in_blocks._read_remote_args
-        blocks = []
+        read_tasks = []
         metadata = []
+        # TODO(Clark): Add CreationStage so we're not shoe-horning read tasks into a
+        # BlockList.
         for i, read_task in enumerate(self._in_blocks._read_tasks):
-            blocks.append(ray.put([read_task]))
+            read_tasks.append(read_task)
```

Comment:
Why change this?

Reply:
The fused read stage isn't out-of-band serializable if it contains Ray object references.